Added NRF52 library (added examples are originally for PlatformIO)

aiunderstand opened the pull request:
Thanks to the great work of @tipih and @sandeepmistry, with very minor changes (only substituting header files for the nrf52) this library enables radio communication between Microbit V1 (using nrf51 lib) and V2 (using nrf52 lib).

Added an example how to use it with V1 as controller (buttons send control message) and V2 as receiver (process the send message). Just like with the nrf51 examples, it requires an Arduino.h (dependency on https://sandeepmistry.github.io/arduino-nRF5/package_nRF5_boards_index.json). 

This library has not been battletested nor used for long durations. Feedback appreciated. Also addresses #1.

aiunderstand committed:
Added NRF52 library (added examples are originally made for PlatformIO hence requires Arduino.h)

aiunderstand committed:
Update README.md

aiunderstand committed:
Update README.md

Faryour commented:
Hello @aiunderstand, I tried using the NRF52 library but had no success. Attached below is the error I am getting on Arduino. Can you be of help, please?
![error](https://user-images.githubusercontent.com/23299593/208233438-98c5a364-c710-44f0-a8c3-74ebb68ccc3a.png)

aiunderstand committed:
Removed reference to nrf52.h, added .ino example for MicroBit v2

aiunderstand committed:
Merge branch 'master' of https://github.com/aiunderstand/NRF52_Radio_library

aiunderstand commented:
Hi @Faryour, thanks for reporting. There were 2 wrong references in nrf52_radio_library.h that I fixed in my PlatformIO project but not committed. Try pulling the new commit. As a bonus, I added a .ino example that should work right away if you have 2 microbit V2's. My settings and serial result:

![image](https://user-images.githubusercontent.com/6376127/208246236-295c12cd-93a6-4df3-9f2b-9ac13bd56b23.png)

![image](https://user-images.githubusercontent.com/6376127/208246273-4e97ef57-0949-43b2-afc6-041c95ee51c0.png)

Faryour commented:
Thank you very much. Tested and working. Appreciated boss.

On Sat, 17 Dec 2022, 15:16 Steven Bos, ***@***.***> wrote:

> Hi @Faryour <https://github.com/Faryour>, thanks for reporting. There
> were 2 wrong references in nrf52_radio_library.h that I fixed in my
> PlatformIO project but not committed. As a bonus, I added a .ino example
> that should work right away if you have 2 microbit V2's. My settings and
> serial result:
>
> [image: image]
> <https://user-images.githubusercontent.com/6376127/208246236-295c12cd-93a6-4df3-9f2b-9ac13bd56b23.png>
>
> [image: image]
> <https://user-images.githubusercontent.com/6376127/208246273-4e97ef57-0949-43b2-afc6-041c95ee51c0.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tipih/NRF51_Radio_library/pull/2#issuecomment-1356276669>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFRYMCNUQBTF3LT2AZZ5OK3WNXDKVANCNFSM537CDIEA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
